dali/test/dali_test_config.cc

```diff
@@ -36,11 +38,33 @@ const std::string &dali_extra_path() {
   std::call_once(noninit_warning, []() {
       auto ptr = std::getenv("DALI_EXTRA_PATH");
       if (!ptr) {
-        std::cerr << "DALI_EXTRA_PATH not initialized." << std::endl;
+        std::cerr << "WARNING: DALI_EXTRA_PATH not initialized." << std::endl;
         *_dali_extra_path = ".";
       } else {
         *_dali_extra_path = std::string(ptr);
       }
+      auto check_hash = "cd " + *_dali_extra_path + " ; git rev-parse HEAD";
+      constexpr int kHashLen = 40;
+      char hash[kHashLen];
+      auto pipe = popen(check_hash.c_str(), "r");
+      if (pipe == NULL) {
+        std::cerr << "WARNING: Could not read the sha of DALI_extra at " << *_dali_extra_path
+                  << ". Make sure DALI_EXTRA is checked out to " << DALI_EXTRA_TAG
+                  << " so the test data is in the required version." << std::endl;
+        return;
+      }
+      auto count = fread(hash, sizeof(hash[0]), kHashLen, pipe);
+      auto ret = pclose(pipe);
+      if (ret != 0 || count != kHashLen) {
+        std::cerr << "WARNING: Could not read the sha of DALI_extra at " << *_dali_extra_path
+                  << ". Make sure DALI_EXTRA is checked out to " << DALI_EXTRA_TAG
+                  << " so the test data is in the required version." << std::endl;
+        return;
+      }
+      if (strncmp(DALI_EXTRA_TAG, hash, kHashLen) != 0) {
+        std::cerr << "Error: DALI_extra at " << *_dali_extra_path << " is not checked out to "
```

Comment:
Why `Error`, not `WARNING`? You let the tests run...
I'd rephrase the message to:
```
WARNING: DALI_extra (<path here>) is in different version than one required by the test. Some tests mail fail to run or produce incorrect results.
Required version: <hash>
Version found: <hash>
```

Reply:
Done